collect/cache/cache.go

```diff
@@ -23,6 +24,8 @@ type Cache interface {
 	TakeExpiredTraces(now time.Time) []*types.Trace
 }
 
+var _ Cache = (*DefaultInMemCache)(nil)
+
 // DefaultInMemCache keeps a bounded number of entries to avoid growing memory
 // forever. Traces are expunged from the cache in insertion order (not access
```

Comment:
There are now a lot of misleading comments in this file -- some are from this PR, some are older than that (like this one, which is no longer quite true and hasn't been for a while). 

Could you try to clean things up a bit? There are a couple of comments about thread safety, a couple about the circular buffer.